scripts/changesummary.py

```diff
@@ -0,0 +1,540 @@
+        """
+        # The paths of the 2 discovery artifacts to compare
+        current_artifact_path = os.path.join(self._current_artifacts_dir, filename)
+        new_artifact_path = os.path.join(self._new_artifacts_dir, filename)
+
+        # Use a helper functions to load the discovery artifacts into pandas
+        # dataframes
+        current_doc = self._load_json_to_dataframe(current_artifact_path)
+        new_doc = self._load_json_to_dataframe(new_artifact_path)
+
+        # Concatenate the 2 dataframes, transpose them, and create
+        # a new dataframe called combined_docs with columns
+        # `Keys`, `CurrentValue`, `NewValue`.
+        combined_docs = (
+            pd.concat([current_doc, new_doc], keys=["CurrentValue", "NewValue"])
+            .transpose()
+            .rename_axis("Key")
+            .reset_index()
+        )
+
+        # This line is for cleanup purposes. As a result of the previous code
+        # we have a multi-index which is not needed so we need to drop a level.
+        combined_docs.columns = combined_docs.columns.droplevel(1)
+
+        # When discovery documents are added, the column `CurrentValue` will
+        # not exist. In that case, we'll just populate with `np.nan`.
+        if "CurrentValue" not in combined_docs.columns:
+            combined_docs["CurrentValue"] = np.nan
+
+        # When discovery documents are deleted, the column `NewValue` will
+        # not exist. In that case, we'll just populate with `np.nan`.
+        if "NewValue" not in combined_docs.columns:
+            combined_docs["NewValue"] = np.nan
+
+        # Split the Key into 2 columns for `Parent` and `Child` in order
+        # to group keys with the same parents together to summarize the changes
+        # by parent.
+        parent_child_df = combined_docs["Key"].str.rsplit(".", 1, expand=True)
+        # Rename the columns and join them with the combined_docs dataframe.
+        # If we only have a `Parent` column, it means that the Key doesn't have
+        # any children.
+        if len(parent_child_df.columns) == 1:
+            parent_child_df.columns = ["Parent"]
+        else:
+            parent_child_df.columns = ["Parent", "Child"]
+        combined_docs = combined_docs.join(parent_child_df)
+
+        # Create a new column `Added` to identify rows which have new keys.
+        combined_docs["Added"] = np.where(
+            combined_docs["CurrentValue"].isnull(), True, False
+        )
+
+        # Create a new column `Deleted` to identify rows which have deleted keys.
+        combined_docs["Deleted"] = np.where(
+            combined_docs["NewValue"].isnull(), True, False
+        )
+
+        # Aggregate the keys added by grouping keys with the same parents
+        # together to summarize the changes by parent rather than by key.
+        parent_added_agg = (
+            combined_docs.groupby("Parent")
+            .Added.value_counts(normalize=True)
+            .reset_index(name="Percent")
```

Comment:
Unless we multiply by 100, isn't this a proportion rather than a percent?

Reply:
Good catch! Fixed in https://github.com/googleapis/google-api-python-client/pull/1286/commits/fa13ac3b7fba388737a09e380b6de8a279b04f5b.